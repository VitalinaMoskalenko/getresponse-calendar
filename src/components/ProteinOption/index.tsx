import React from "react";
import styled from "styled-components";
import Button from "../Button";
import { SmallBody } from "../Paragraphs";

type ImgPropsType = {
  iconUrl: string;
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0px;
`;

const IconContainer = styled.div`
  margin-top: 5px;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 0px;
`;

const Img = styled.img<ImgPropsType>`
  content: url(${({ iconUrl }) => `${iconUrl}`});
  margin-right: 10px;
  height: 35px;
`;

type PropsType = {
  icon: string;
  text?: string;
  onClick: () => void;
};

const ProteinOption = ({ text, icon, onClick }: PropsType) => {
  return (
    <Container>
      <SmallBody>{text}</SmallBody>
      <IconContainer>
        <Img iconUrl={icon} onClick={onClick} />
        <Img iconUrl={icon} onClick={onClick} />
        <Img iconUrl={icon} onClick={onClick} />
        <Img iconUrl={icon} onClick={onClick} />
        <Img iconUrl={icon} onClick={onClick} />
      </IconContainer>
    </Container>
  );
};

export default ProteinOption;
