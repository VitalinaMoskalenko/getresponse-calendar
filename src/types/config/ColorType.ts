export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
