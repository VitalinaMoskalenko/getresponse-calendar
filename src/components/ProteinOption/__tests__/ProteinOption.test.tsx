import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import ProteinOption from "../";
import { theme } from "../../../config";

it("renders ProteinOption correctly", () => {
  const proteinOptionComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ProteinOption text="" icon="" onClick={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(proteinOptionComponent).toMatchSnapshot();
});
