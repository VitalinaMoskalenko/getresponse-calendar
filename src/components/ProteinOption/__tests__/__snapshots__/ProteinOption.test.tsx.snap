// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProteinOption correctly 1`] = `
<div
  className="sc-dlnjwi hqeLaF"
>
  <p
    className="sc-gtsrHT fsyTPD"
  >
    
  </p>
  <div
    className="sc-hKFxyN gZrZNh"
  >
    <img
      className="sc-eCApnc cWOuTk"
      onClick={[Function]}
    />
    <img
      className="sc-eCApnc cWOuTk"
      onClick={[Function]}
    />
    <img
      className="sc-eCApnc cWOuTk"
      onClick={[Function]}
    />
    <img
      className="sc-eCApnc cWOuTk"
      onClick={[Function]}
    />
    <img
      className="sc-eCApnc cWOuTk"
      onClick={[Function]}
    />
  </div>
</div>
`;
